clang/test/SemaCXX/union-member-destructor.cpp

```diff
@@ -0,0 +1,48 @@
+// RUN: %clang_cc1 -std=c++11 -fsyntax-only -verify %s
+
+namespace t1{
+template <class T> struct VSX {
+  ~VSX() { static_assert(sizeof(T) != 4, ""); } // expected-error {{static assertion failed due to requirement 'sizeof(int) != 4':}} \
+                                                // expected-note {{expression evaluates to '4 != 4'}}
+};
+struct VS {
+  union {
+    VSX<int> _Tail;
+  };
+  ~VS() { }
+  VS(short);
+};
+VS::VS(short) : _Tail() { } // expected-note {{in instantiation of member function 't1::VSX<int>::~VSX' requested here}}
```

Comment:
Based on https://eel.is/c++draft/class.base.init#12 this should not apply to delegating constructors, can we add a test if once does not exist that verifies this.

Reply:
I added a delegating constructor to the test. As the delegating constructor invokes the one that triggers the assertion, the assertion has to come either way. Therefore, I assume the test should assert that the note does not appear at the delegating constructor.

Reply:
Makes sense, that might be a bit subtle for folks running into the test, so it is worth it to add a comment noting that.

Reply:
alright, done.